src/libraries/System.Net.Mail/tests/Functional/FakeSmtpServer.cs

```diff
@@ -0,0 +1,265 @@
+using System;
```

Comment:
What is the difference in purpose between this FakeSmtpServer.cs and the SmtpServer.cs we already have in tests?
https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Mail/tests/Functional/SmtpServer.cs
Why do we need both?

Reply:
I see, you're deleting that one and creating this "Fake" one.  Why is this called "Fake"?  Isn't it functional enough for SmtpClient to connect to it?  Should it be called LoopbackSmtpServer or something like that?

Reply:
SmtpServer is replaced by FakeSmtpServer. The diff is more exposed properties related to Auth and the ability to block at specific of the request (to enable tests for cancellation)

Reply:
The name is up for discussion :) see comment about it above

Reply:
>The name is up for discussion :) see comment about it above

>Should it be called LoopbackSmtpServer or something like that?

I initially started this discussion reacting to the word "mock" vs. "fake".
 
But actually, in thinking about this more, I tend to agree that this is a real server and not really a "fake". It just has, perhaps, limited functionality which we might grow depending on our needs for more complex tests. It is no different than our other loopback test servers we have for http and proxy use.

Much of this SmtpClient code and tests were authored before we standarized our naming conventions for things like this. So, I would support naming this as 'LoopbackSmtpServer'.